pymeasure/adapters/adapter.py

```diff
@@ -33,48 +37,128 @@ class Adapter:
 
     This class should only be inherited from.
 
-    :param preprocess_reply: optional callable used to preprocess strings
-        received from the instrument. The callable returns the processed string.
+    :param preprocess_reply: optional callable used to preprocess
+        strings received from the instrument. The callable returns the
+        processed string.
+
+        .. deprecated :: 0.10
+            Implement it in the instrument's `read` method instead.
+
+    :param log: Parent logger of the 'Adapter' logger.
     :param kwargs: all other keyword arguments are ignored.
     """
 
-    def __init__(self, preprocess_reply=None, **kwargs):
+    def __init__(self, preprocess_reply=None, log=None, **kwargs):
         self.preprocess_reply = preprocess_reply
         self.connection = None
+        if log is None:
+            self.log = logging.getLogger("Adapter")
+        else:
+            self.log = log.getChild("Adapter")
+        self.log.addHandler(logging.NullHandler())
```

Comment:
If you use several devices at the same time, all the different Adapter Loggers would have the same name. In this case, the Adapter has the name of the instrument (module) it is called from, such that you can distinguish the communication of different devices.

Reply:
Ah, I just scratched my head how that works, then I realised it's a manual thing, not automatic! ^.^

Reply:
Yes, if you manually give the `Adapter` a logger, it will create a child with the name "Adapter" in order to distinguish it from the instrument logger. If you do not care, you get a default one similar to the previous behavior.